subsys/bluetooth/services/lbs.c

```diff
@@ -46,8 +46,19 @@ static ssize_t write_led(struct bt_conn *conn,
 	LOG_DBG("Attribute write, handle: %u, conn: %p", attr->handle,
 		(void *)conn);
 
+	if (len != 1) {
+		LOG_DBG("Write led: Incorrect data length");
+		return len;
+	}
+
 	if (lbs_cb.led_cb) {
-		lbs_cb.led_cb(*(bool *)buf);
+		uint8_t val = *((uint8_t *)buf);
+
+		if (val == 0x00 || val == 0x01) {
+			lbs_cb.led_cb(val ? true : false);
+		} else {
+			LOG_DBG("Write led: Incorrect value");
```

Comment:
Seems more appropriate to return BT_ATT_ERR_VALUE_NOT_ALLOWED in this case.

Reply:
Sure. Fixed.